primitives/runtime/src/lib.rs

```diff
@@ -83,6 +90,53 @@ pub const ROOT_ACCOUNT_DERIVATION_PREFIX: &[u8] = b"pallet-bridge/account-deriva
 #[derive(RuntimeDebug, Default, Clone, Copy, Eq, Hash, PartialEq)]
 pub struct HeaderId<Hash, Number>(pub Number, pub Hash);
 
+/// Generic header id provider.
+pub trait HeaderIdProvider<Number, Hash> {
+	// Get the header id.
+	fn id(&self) -> HeaderId<Hash, Number>;
+
+	// Get the header id for the parent block.
+	fn parent_id(&self) -> Option<HeaderId<Hash, Number>>;
+}
+
+impl<Number, Hash> HeaderIdProvider<Number, Hash::Output> for Header<Number, Hash>
```

Comment:
I was thinking of something like `impl<T> HeaderIdProvider for T where T: sp_runtime::traits::Header` instead of relying on `generic::Header`. Then we don't need this extra `+ HeaderIdProvider<Self::BlockNumber, Self::Hash>` requirement.

Can you, please, check if it works? I may be missing something :) Thanks!

Reply:
Oh sorry, total miss on my side. Done. Looks much better.

Reply:
That's fine, thank you :)